notebooks/20220330_correlations_vstnorm.ipynb

```diff
@@ -0,0 +1,537 @@
+    "# write normalized df to file\n",
+    "write_csv(vstnorm_pa14_kmers, \"sandbox/202203_normalize/pa14_vstnorm_kmers.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cfa5318f",
+   "metadata": {},
+   "source": [
+    "## Read in hash map and label with operons"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "e841ca0b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pa14_hash_map <- read_csv(\"outputs/txomes_sourmash_sketch_singleton/pa14.csv\", show_col_types = F)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "95b049d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pa14_operons <- read_csv(\"https://raw.githubusercontent.com/greenelab/core-accessory-interactome/master/data/metadata/PA14-operons-2021-07-19.csv\", show_col_types = F)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "caa29b21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pa14_transcript_to_gene <- read_csv(\"https://osf.io/ema5c/download\", skip = 1, col_names = c(\"tmp\", \"transcript_name\", \"locus_tag\"), show_col_types = F) %>%\n",
+    "  select(-tmp)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "fc055ed4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pa14_hash_map <- pa14_hash_map %>%\n",
+    "  mutate(transcript_name = gsub(\" .*\", \"\", name)) %>%\n",
+    "  left_join(pa14_transcript_to_gene, by = \"transcript_name\") %>%\n",
+    "  left_join(pa14_operons, by = \"locus_tag\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "1321d7a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pa14_hash_map_operon <- pa14_hash_map %>%\n",
+    "  filter(!is.na(`operon-id`))\n",
+    "\n",
+    "pa14_hash_map_operon_small <- pa14_hash_map_operon %>%\n",
+    "  select(minhash, operon_id = `operon-id`) %>%\n",
+    "  distinct()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd5d4756",
+   "metadata": {},
+   "source": [
+    "## Filter our \"rare\" k-mers\n",
+    "\n",
+    "DESeq2 normalization was done with minimal filtering, keeping all k-mers that were in at least 10 data sets. \n",
+    "Further filter out \"rare\" k-mers so that corelations can be performed without exhausting vector memory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "1dff999d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "7601"
+      ],
+      "text/latex": [
+       "7601"
+      ],
+      "text/markdown": [
+       "7601"
+      ],
+      "text/plain": [
+       "[1] 7601"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# define a vector of kmers to keep; limit to ones that are present in many samples\n",
+    "keep_kmers <- rownames(raw_pa14_kmers_pa)[rowSums(raw_pa14_kmers_pa) > 100]\n",
```

Comment:
in other notebooks you use > 50, why do you choose > 100 here?

Reply:
this one was more compute intensive for some reason and kept running out of ram unless i was more stringent with filtering :( I may try and run it again with 50 on a different computer